src/natcap/invest/recreation/recmodel_server.py

```diff
@@ -589,7 +622,8 @@ def _populate_offset_queue(block_offset_size_queue):
         n_points = 0
 
         while True:
-            point_array = numpy_array_queue.get()
+            # if the queue item is 'STOP', this returns 'STOP', not an array.
+            point_array = _numpy_loads(numpy_array_queue.get())
```

Comment:
If you don't mind, I prefer a little more explicit check when it comes to having sentinels in a queue before trying to process the result. Something like this:

```
payload = numpy_array_queue.get()
if payload == 'STOP':
    n_parse_processes -= 1
    if n_parse_processes == 0:
        break
    continue
else:
    point_array = _numpy_loads(payload)
    ....
```

Then you won't have to have `_numpy_loads` doing double duty by also having to know the context in which the data come in, you can just have it load blindly.

Reply:
You're absolutely right! not sure why I did it that way